src/fastcs/connections/ip_connection.py

```diff
@@ -16,39 +52,43 @@ class IPConnection:
     def __init__(self):
         self._reader, self._writer = (None, None)
         self._lock = asyncio.Lock()
+        self.connected: bool = False
+
+    @property
+    def lock(self) -> asyncio.Lock:
+        return self._lock
```

Comment:
This should be removed if `_with_lock` is removed.

Reply:
Originally I didn't have this property, I only created it as the linter complained. The intention is that this would only be used in the decorator so wrap the function in the context manager, not use outside the class.